src/session/update-user.ts

```diff
@@ -26,18 +26,18 @@ export type UpdateUser = (
   req: IncomingMessage | NextApiRequest,
   res: ServerResponse | NextApiResponse,
   user: Claims
-) => void;
+) => Promise<void>;
 
 /**
  * @ignore
  */
 export default function updateUserFactory(sessionCache: SessionCache): UpdateUser {
-  return (req, res, user): void => {
-    sessionCache.init(req, res, false);
-    const session = sessionCache.get(req, res);
+  return async (req, res, user) => {
+    await sessionCache.init(req, res, false);
+    const session = await sessionCache.get(req, res);
     if (!session || !user) {
       return;
     }
-    sessionCache.set(req, res, { ...session, user });
+    await sessionCache.set(req, res, { ...session, user });
```

Comment:
Is this last `await` necessary, or could it be just `return sessionCache.set(req, res, { ...session, user });`?

Reply:
You could do either. It only really makes a difference when you're using try/catch statements

```js
try {
  return promise;
} catch () {}

// is very different from
try {
  return await promise;
} catch () {}
```

Returning from this would change the function return type from `Promise<void>` to Promise<undefined>`, I think for a function that shouldn't return anything `Promise<void>` is probably more suitable